File: 2015/01/01.kt
import java.io.File

fun part01(input: String): Int {
    var counter = 0
    input.forEach {
        if (it == '(') {
            counter++
        }
        if (it == ')') {
            counter--
        }
    }
    return counter
}

fun part02(input: String): Int {
    var counter = 0
    input.forEachIndexed { index, it ->
        if (it == '(') {
            counter++
        }
        if (it == ')') {
            counter--
        }
        if (counter < 0) {
            return index + 1
        }
    }
    return 0
}

fun main() {
    val inputString = File("input.txt").readText()
    println("Part 01: ${part01(inputString)}")
    println("Part 02: ${part02(inputString)}")
}

File: 2015/01/01.rb

def part01(input)
    counter = 0
    input.each_char do |char|
        counter += 1 if char == "("
        counter -= 1 if char == ")"
    end
    return counter
end

def part02(input)
    counter = 0
    input.length.times do |index|
        counter += 1 if input[index] == "("
        counter -= 1 if input[index] == ")"
        return index + 1 if counter < 0
    end
    return counter
end

File.open("input.txt") do |file|
    Input = file.read
    puts "Part01: #{part01(Input)}"
    puts "Part02: #{part02(Input)}"
end
